database/database_test.go

```diff
@@ -234,3 +237,412 @@ func TestStatesWithStatus(t *testing.T) {
+		t.Errorf("[duplicate domains] cannot put states %s", err)
+		return
+	}
+}
+
+// TestGetDuplicateIneligibleDomainStates is testing getting duplicate
+// IneligibleDomainStates from the database
+func TestGetDuplicateIneligibleDomainStates(t *testing.T) {
+	resetDB()
+
+	// add domains to the database
+	for _, tt := range setIneligibleDomainTests {
+
+		var statuses []string
+		statusReport := func(format string, args ...interface{}) {
+			formatted := fmt.Sprintf(format, args...)
+			statuses = append(statuses, formatted)
+		}
+
+		err := testDB.SetIneligibleDomainStates(
+			tt.domainStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+
+		if !reflect.DeepEqual(statuses, tt.wantStatusReports) {
+			t.Errorf("[%s] Incorrect status reports: %#v", tt.description, statuses)
+		}
+	}
+
+	// get duplicate domains from the database
+	var domainNames = []string{"gmail.test", "gmail.test"}
+
+	domainStates, err := testDB.GetIneligibleDomainStates(domainNames)
+	if err != nil {
+		t.Errorf("%s", err)
+	}
+
+	sort.Slice(domainStates, func(i, j int) bool { return domainStates[i].Name < domainStates[j].Name })
+	if len(domainStates) != len(domainNames) {
+		t.Errorf("Incorrect count of states for test getDuplicateDomains")
+	}
+	for i, domainState := range domainStates {
+		if domainState.Name != (domainNames[i]) {
+			t.Errorf("unexpected domain at position %d for test getDuplicateDomains: %#v", i, domainState)
+		}
+	}
+}
+
+// TestDeleteDuplicateIneligibleDomainStates is testing deleting duplicate
+// IneligibleDomainStates from the database
+func TestDeleteDuplicateIneligibleDomainStates(t *testing.T) {
+	resetDB()
+
+	// add domains to the database
+	var statuses []string
+	statusReport := func(format string, args ...interface{}) {
+		formatted := fmt.Sprintf(format, args...)
+		statuses = append(statuses, formatted)
+	}
+	for _, tt := range setIneligibleDomainTests {
+		err := testDB.SetIneligibleDomainStates(
+			tt.domainStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+	}
+
+	// delete domains from the database
+	var domainNames = []string{"gmail.test", "gmail.test"}
+	err := testDB.DeleteIneligibleDomainStates(domainNames)
+	if err != nil {
+		t.Errorf("%s", err)
+	}
+
+	// get domains from the database
+	// should not exist as they are deleted
+	var domainStates []IneligibleDomainState
+	for _, tr := range getAndDeleteTests {
+		domainStates, err = testDB.GetIneligibleDomainStates(tr.domainNames)
+
+		if merr, ok := err.(datastore.MultiError); ok {
+			for _, err := range merr {
+				if err != datastore.ErrNoSuchEntity {
+					t.Errorf("%s", err)
+				}
+				_ = err
+			}
+		} else {
+			if err != datastore.ErrNoSuchEntity {
+				t.Errorf("%s", err)
+			}
+		}
+	}
```

Comment:
This only needs to check that that the domain you deleted is no longer in the database.